sn_cli/src/subcommands/register.rs

```diff
@@ -138,3 +145,15 @@ async fn get_registers(names: Vec<String>, client: &Client) -> Result<()> {
 
     Ok(())
 }
+
+/// Parse str and return the address and the register info for printing
+fn parse_addr(str: &str, use_name: bool, pk: PublicKey) -> Result<(RegisterAddress, String)> {
```

Comment:
here the returned `String` is for `register info` , but above `String` is for `address`
better make all this `type safe`, i.e.  using type such as `RegisterAddress, RegisterInfo, RegisterName, ......` to avoid confusion or misleading.

Reply:
Renamed the source of the confusion, which was the user-input `address_str` that I renamed as such.